### DISCUSSION TITLE
CVE-2021-44228 Advisory

### DISCUSSION THREAD

**Discussion by mih-kopylov** [community/community#24647] [Code Security]
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/advisories/GHSA-jfh8-c2jp-5v3q">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/advisories/GHSA-jfh8-c2jp-5v3q" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/6898003/181085822-251c763f-23ed-4abd-9b22-52555312cf48.png" class="thumbnail onebox-avatar" width="240" height="240">

<h3><a href="https://github.com/advisories/GHSA-jfh8-c2jp-5v3q" target="_blank" rel="noopener nofollow ugc">CVE-2021-44228 - GitHub Advisory Database</a></h3>

  <p>Remote code injection in Log4j</p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
The advisory says that both log4j-core and log4j-api are affected.</p>
<p>According to <a href="https://nvd.nist.gov/vuln/detail/CVE-2021-44228" class="inline-onebox" rel="noopener nofollow ugc">NVD - CVE-2021-44228</a> the <code>log4j-api</code> isn’t affected, it’s only <code>log4j-core</code> that is.</p>
<p><a href="https://issues.apache.org/jira/browse/LOG4J2-3201?focusedCommentId=17456962&amp;page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17456962" class="onebox" target="_blank" rel="noopener nofollow ugc">https://issues.apache.org/jira/browse/LOG4J2-3201?focusedCommentId=17456962&amp;page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17456962</a><br>
The log4j devs confirm that.</p>
<p>I’m not sure where does the information about <code>log4j-api</code> comes to database, could it be verified?</p>

**Comment by murphy85** [comment]
<p>The advisory was updated with an explanation, but I’m not happy with the solution, too.</p>
<blockquote>
<p>Only the <code>org.apache.logging.log4j:log4j-core</code> package is directly affected by this vulnerability. The <code>org.apache.logging.log4j:log4j-api</code> package is included on this advisory to prevent version desync which may occur as a result of dependabot actions.</p>
</blockquote>
<p>Our security scanners, like trivy, alerts us on EVERY Spring Boot project (over 100), because they depend on log4j-api, but NOT on log4j-core. Therefore they are not affected by this CVE.</p>

**Comment by boerpb** [comment]
<p>Dependency version desync is not a valid reason for a package to be included as a security risk.</p>
<p>Considering the impact this might have in automated CI/CD pipelines, where non-passing security tests usually do not pass quality gates and devs are therefore more likely to add an exception to security scanners (thus creating a real security risk), the <code>log4j-api</code> dependency should be removed from the vuln database asap.</p>

**Comment by clamoriniere** [comment]
<p>Hello</p>
<p>it seems that all the fixed versions are not listed for the packages: <code>log4j-core</code> <code>log4j-api</code></p>
<ul>
<li>
<code>2.12.2</code> log4j version should be listed with “2.15.0” as “Fixed version” for <code>CVE-2021-44228</code>
</li>
</ul>
<p>Same issue about another log4j vulnerability CVE-2021-45046:</p>
<ul>
<li>
<code>2.12.2</code> log4j version should be listed with “2.16.0” as “Fixed version” for <code>CVE-2021-45046</code>
</li>
</ul>
<p>How can we add them?</p>

**Comment by MikeMoore63** [comment]
<p>Any progress on removing the false positive for log4j-api from this advisory?</p>

**Comment by afdesk** [comment]
<p>I took a look at <a href="https://nvd.nist.gov/vuln/detail/CVE-2021-44228" rel="noopener nofollow ugc">NVD Details about CVE-2021-44228</a>, and there are security releases 2.12.2, 2.12.3, and 2.3.1.</p>
<p>But GitHub Security Advisory Database doesn’t exclude these versions: <a href="https://github.com/advisories/GHSA-jfh8-c2jp-5v3q" class="inline-onebox" rel="noopener nofollow ugc">Remote code injection in Log4j · CVE-2021-44228 · GitHub Advisory Database · GitHub</a></p>
<p>Is it possible to fix it?<br>
thanks a lot!</p>

**Comment by mrjonstrong** [comment, marked as answer]
<p>advisory has been updated now :smile:</p>

**Comment by jcburgoon** [comment]
<p>The update does not fix the issue since it does not take all Java versions into account. See <a href="https://github.com/advisories/GHSA-8489-44mv-ggj8" class="inline-onebox" rel="noopener nofollow ugc">Improper Input Validation and Injection in Apache Log4j2 · CVE-2021-44832 · GitHub Advisory Database · GitHub</a> for an example.</p>
<p>The same problem applies to <a href="https://github.com/advisories/GHSA-7rjr-3q55-vv33" class="inline-onebox" rel="noopener nofollow ugc">Incomplete fix for Apache Log4j vulnerability · CVE-2021-45046 · GitHub Advisory Database · GitHub</a></p>
